core/modules/catalog/index.ts

```diff
@@ -17,4 +20,15 @@ export const CatalogModule: StorefrontModule = function (app, store, router, mod
   store.registerModule('stock', stockModule)
   store.registerModule('tax', taxModule)
   store.registerModule('category', categoryModule)
+
+  // Things moved from Product.js
+  EventBus.$on('product-after-priceupdate', product => productAfterPriceupdate(product, store))
```

Comment:
@lukeromanowicz good point! we need to have them `$off` somewhere OR just bind in the CSR. Going this way - these events are used because of how the theme works. It would be probably better to move them out from Vuex from the architecture point of view I guess. I'm not sure about this second statement as they're in the end not touching UI directly (but working on data models; therefore it was fine with me to keep it here) but for sure we need to take care of `off`

Reply:
good point - on client-side it's invoked once, i'll change this not to be invoked on Server side :)